teuthology/orchestra/remote.py

```diff
@@ -458,6 +458,131 @@ def get_tar_stream(self, path, sudo=False):
             ])
         return self.run(args=args, wait=False, stdout=run.PIPE)
 
+    def copy_file(self, src, dst, sudo=False, mode=None, owner=None,
+                                              mkdir=False, append=False):
+        """
+        Copy data to remote file
+
+        :param src:     source file path on remote host
+        :param dst:     destination file path on remote host
+        :param sudo:    use sudo to write file, defaults False
+        :param mode:    set file mode bits if provided
+        :param owner:   set file owner if provided
+        :param mkdir:   ensure the destination directory exists, defaults False
+        :param append:  append data to the file, defaults False
+        """
+        dd = 'sudo dd' if sudo else 'dd'
+        args = dd + ' if=' + src + ' of=' + dst
+        if append:
+            args += ' conv=notrunc oflag=append'
+        if mkdir:
+            mkdirp = 'sudo mkdir -p' if sudo else 'mkdir -p'
+            dirpath = os.path.dirname(dst)
+            if dirpath:
+                args = mkdirp + ' ' + dirpath + '\n' + args
+        if mode:
+            chmod = 'sudo chmod' if sudo else 'chmod'
+            args += '\n' + chmod + ' ' + mode + ' ' + dst
+        if owner:
+            chown = 'sudo chown' if sudo else 'chown'
+            args += '\n' + chown + ' ' + owner + ' ' + dst
+        args = 'set -ex' + '\n' + args
+        self.run(args=args)
```

Comment:
i'd suggest rewrite this using a helper function for adding optional `sudo` and f-string, but my suggestion was dropped by accident when posting it. i will add a follow-up PR later on.

Reply:
thanks for the heads up, I also would like to backport this patch to py2, however I'm not sure that I will do that, hopefully we will not need py2 based branches of teuthology downstream in couple of months.